### PR TITLE
remove sqlite use, baby steps

### PR DESCRIPTION
**Pull request by Villemoes**
The most significant part of these patches is that they cut "building dependency tree" for oe bake world from 1m39s to about 18s. Aside from that, the diffstat is also quite nice; about 100 lines removed from files that are still used.

One could certainly argue that abe0274 and d59ab1a should be squashed, but I'd like to keep a record of how I got there.

### CODE REVIEW

File: lib/oelite/runq.py [hunk 1]
```diff
@@ -352,7 +352,7 @@ def resolve_dependency(item, recursion_path, deptype):
             recursion_path[1].append(str(item))
 
             # try to get cached recdepends list of packages
-            packages = self.get_recdepends(package, deptype)
```
Comment (kimrhh) [inline lib/oelite/runq.py]: shouldn't the get_recdepends method in runq die then?

**Comment by kimrhh** [comment]
looks good to me, if we can actually demonstrate it does not make the builds splat :)

File: lib/oelite/runq.py (hunk 1 above)
Comment (Villemoes) [inline lib/oelite/runq.py]: Isn't that what e173380 does?

Comment (kimrhh) [inline lib/oelite/runq.py]: oh, indeed. :+1:

**Pull request merged**